src/optimistictransactiondb.rs

```diff
@@ -0,0 +1,229 @@
+use crate::{ColumnFamily, ColumnFamilyDescriptor, Error, Options, Transaction, WriteOptions, DB};
+use ffi;
+use libc::{c_int, c_uchar};
+use std::collections::BTreeMap;
+use std::ffi::CString;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+use std::ptr;
+use std::str;
+use std::sync::{Arc, RwLock};
+
+pub struct OptimistictransactionDB {
```

Comment:
Probably use nicer casing `OptimisticTransactionDB`? (and for other similar names)